app/main.py

```diff
@@ -1 +1,22 @@
-# write your code here
+from __future__ import annotations
+
+from app.players.dwarves.dwarf_warrior import DwarfWarrior
+from app.players.dwarves.dwarf_blacksmith import DwarfBlacksmith
+from app.players.elves.druid import Druid
+from app.players.elves.elf_ranger import ElfRanger
+
+
+def calculate_team_total_rating(team: list[Druid | ElfRanger]) -> int:
+    return sum(elf_player_class.get_rating() for elf_player_class in team)
+
+
+def elves_concert(elves: list[ElfRanger | Druid]) -> None:
+    for elf in elves:
+        elf.play_elf_song()
+
+
+def feast_of_the_dwarves(
```

Comment:
list[Dwarf]

Reply:
Thanks for attention to ABC inheritance, I misunderstood description=) Also I made mistake with Player annotation at rating_calculation_function as you pointed. I think other annotations are fine because at readme we are passing lists of children instances not Dwarf and Elf